ir/constant/const_float.go

```diff
@@ -101,7 +102,24 @@ func NewFloatFromString(typ *types.FloatType, s string) (*Float, error) {
 			x, nan := f.Big()
 			return &Float{Typ: typ, X: x, NaN: nan}, nil
 		case strings.HasPrefix(s, "0xM"):
-			//s = s[len("0xM"):]
+			// From https://llvm.org/docs/LangRef.html#simple-constants
+			//
+			// > The 128-bit format used by PowerPC (two adjacent doubles) is represented by 0xM followed by 32 hexadecimal digits.
+			hex := s[len("0xM"):] // first remove the prefix
```

Comment:
Nit: 
```go
hex := strings.TrimPrefix(s, "0xM")
```

(and then it doesn't need the comment)

Reply:
Sounds good, just let you know that's because in this function use `s[len(prefix):]` rather than `TrimPrefix` all places if could I would like to change all of them, are you ok with that? Else I would only change this one for now.

Reply:
> Sounds good, just let you know that's because in this function use s[len(prefix):] rather than TrimPrefix all places if could I would like to change all of them, are you ok with that? Else I would only change this one for now.

Feel free to replace all of them. I agree that `strings.TrimPrefix` is better to use than `s[len("foo"):]`, I had just forgotten about that function before.